portfolio/src/main/webapp/mapScript.js

```diff
@@ -167,9 +167,103 @@ function initMap() {
   map.mapTypes.set('styled_map', styledMapType);
   map.setMapTypeId('styled_map');
 
-  // Set markers on the map.
-  let markerJapanTemple = new google.maps.Marker({position: japanTemple, map: map});
-  let markerNewZealand = new google.maps.Marker({position: newZealandLake, map: map});
-  let markerUganda = new google.maps.Marker({position: ugandaView, map: map});
-  console.log("Initialised the map");
+  // Set markers on the map if marker flag is True.
+  if (markerFlag == true) {
+    let markerJapanTemple = new google.maps.Marker({position: japanTemple, map: map});
+    let markerNewZealand = new google.maps.Marker({position: newZealandLake, map: map});
+    let markerUganda = new google.maps.Marker({position: ugandaView, map: map});
+  }
+  console.log("Initialise new map");
+}
+
+function showEclipse() {
+  // Initialize new map without markers.
+  initMap(false);
+  console.log("Initialise new map");
+  
+  // Parse csv files for solar/lunar eclipses from 2020 to 2030.
+  let solarEclipse = parseCsv(readFile("./eclipse/solar_decade.csv"));
+  console.log("Parse solar eclipses file")
+
+  let lunarEclipse = parseCsv(readFile("./eclipse/lunar_decade.csv"));
+  console.log("Parse lunar eclipses file")
+  
+  // Add markers to the map.
+  for (let i = 1; i < solarEclipse.length; i++) {
+    let lat = parseLat(solarEclipse[i][9]);
+    let lng = parseLng(solarEclipse[i][10]);
+
+    let solarMerker = new google.maps.Marker({
+      position: new google.maps.LatLng(lat, lng), 
+      map: map,
+      icon: {
+        path: google.maps.SymbolPath.CIRCLE,
+        scale: 8.5,
+        fillColor: "#ff3",
+        fillOpacity: 0.4,
+        strokeWeight: 0.4
+      },
+    });
+  }
+  console.log("Add " + (solarEclipse.length - 1).toString() + " solar markers");
+
+  for (let i = 1; i < lunarEclipse.length; i++) {
+    let lat = parseLat(lunarEclipse[i][11]);
+    let lng = parseLng(lunarEclipse[i][12]);
+
+    let lunarMerker = new google.maps.Marker({
+      position: new google.maps.LatLng(lat, lng), 
+      map: map,
+      icon: {
+        path: google.maps.SymbolPath.CIRCLE,
+        scale: 8.5,
+        fillColor: "#491e3c",
+        fillOpacity: 0.4,
+        strokeWeight: 0.4
+      },
+    });
+  }
+  console.log("Add " + (lunarEclipse.length - 1).toString() + " lunar markers");
+}
+
+function readFile(filePath) {
+  let result = null;
+
+  let xmlhttp = new XMLHttpRequest();
+  xmlhttp.open("GET", filePath, false);
+  xmlhttp.send();
+
+  if (xmlhttp.status == 200) {
+    result = xmlhttp.responseText;
+  }
+  return result;
+}
+
+function parseCsv(csvFile) {
```

Comment:
Not necessary to change for this project, but usually it is a bad idea to parse CSV (or other file formats) yourself. It is not used in this case, but CSV supports quoted values that can contain commas that should not be split on.

If CSV data is used for the capstone project, I would recommend doing the parsing on the server with an existing java CSV parsing library and sending the data to the client as JSON.

Reply:
I tried to do that, but wasn't sure how to import third party libraries to the project.

Reply:
You should be able to add the dependency using maven by modifying the `pom.xml` file. E.g. the Apache commons libraries are widely used and well tested.
https://mvnrepository.com/artifact/org.apache.commons/commons-csv/1.8